functions/src/tests/setup.ts

```diff
@@ -0,0 +1,20 @@
+import { TEST_PAYLOAD } from "./utils";
+import * as admin from "firebase-admin";
+import { createUniqueId, SHARE_LINK_FIRESTORE_COLLECTION } from "../utils";
+
+// Test setup run before all tests.
+module.exports = async () => {
+  console.log("Setting up tests...");
+  if (!process.env.FUNCTIONS_EMULATOR) {
+    throw new Error("Test suite must be run with the Firebase emulator.");
+  }
+  // Manually "register" one share link to test against.
```

Comment:
For some of the tests, we need to have a share link already registered (e.g. to see if fetching an existing share link works), so I'm adding one here "by hand" (manually dumping the expected data into the firestore emulator). 

This feels a little fragile because if we're to change how share links are registered (e.g. how `createUniqueId()` works) we will need to know to also update it here.

Some other options could be to:
- Call the `/registerUrl` endpoint here. But, this feels weird as we would be using a function that we're going to be testing in the test setup.
- Rely on the order of the tests to populate firestore (run the `/registerUrl` tests first, then use the data they post in the other tests). But this is generally bad practice

Reply:
I don't have strong feelings, but I'd be fine with using the /registerUrl endpoint.  I think we should still keep an isolated test that tests /registerUrl directly.  But I think it's okay for tests to "build off of" each other and use functionality that's tested in other tests.

+1 to not relying on order of tests.  I don't think we can rely on that, and I often run tests in isolation when testing.  Handy tip in case you haven't come across it: You can change `test(...)` to `test.only(...)` to run a single test.  Though you might also have to change the parent `describe(...)` to `describe.only(...)`

Reply:
Cool, yeah that makes sense. 

I've updated this to not use a dedicated setup (and instead use `beforeAll()` per your suggestion below), and I'm now using `/registerUrl` there